kubernetes.go

```diff
@@ -106,13 +144,31 @@ func (gw *Gateway) RunKubeController(ctx context.Context) error {
 		return err
 	}
 
-	gw.Controller = newKubeController(ctx, kubeClient)
+	gwAPIClient, err := gatewayClient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	gw.Controller = newKubeController(ctx, kubeClient, gwAPIClient)
 	go gw.Controller.run()
 
 	return nil
 
 }
 
+func existGatewayCRDs(ctx context.Context, c *gatewayClient.Clientset) bool {
+
+	_, err := c.GatewayV1alpha2().Gateways("").List(ctx, metav1.ListOptions{})
+	if meta.IsNoMatchError(err) || runtime.IsNotRegisteredError(err) || errors.IsNotFound(err) {
```

Comment:
Should we have added a check on 'forbidden' on here? I just upgraded `k8s_gateway` in my custom coredns build used with the official coredns helm cart and it won't start up because the service account currently doesn't have access to gateways.

I kind of think we should catch the forbidden error and log it with a message saying it doesn't have access (i.e. different from the message here), but probably shouldn't panic when it is forbidden.

```
[WARNING] An external plugin (/go/pkg/mod/github.com/ori-edge/k8s_gateway@v0.2.0/setup.go line 56) is using the deprecated function Normalize. This will be removed in a future versions of CoreDNS. The plugin should be updated to use OriginsFromArgsOrServerBlock or NormalizeExact instead.
[INFO] plugin/k8s_gateway: Building k8s_gateway controller
panic: gateways.gateway.networking.k8s.io is forbidden: User "system:serviceaccount:network:default" cannot list resource "gateways" in API group "gateway.networking.k8s.io" at the cluster scope

goroutine 1 [running]:
github.com/ori-edge/k8s_gateway.existGatewayCRDs({0x214c368, 0xc0001400c8}, 0x1ddc0f4)
    /go/pkg/mod/github.com/ori-edge/k8s_gateway@v0.2.0/kubernetes.go:169 +0x191
github.com/ori-edge/k8s_gateway.newKubeController({0x214c368, 0xc0001400c8}, 0xc00052cf20, 0xc00007bee0)
    /go/pkg/mod/github.com/ori-edge/k8s_gateway@v0.2.0/kubernetes.go:50 +0x11a
github.com/ori-edge/k8s_gateway.(*Gateway).RunKubeController(0xc000176b40, {0x214c368, 0xc0001400c8})
    /go/pkg/mod/github.com/ori-edge/k8s_gateway@v0.2.0/kubernetes.go:152 +0x85
github.com/ori-edge/k8s_gateway.setup(0x1dd8168)
    /go/pkg/mod/github.com/ori-edge/k8s_gateway@v0.2.0/setup.go:29 +0x112
github.com/coredns/caddy.executeDirectives(0xc00021c900, {0x7fff2be57ed8, 0x15}, {0x30d1460, 0x32, 0x1000000004b8066}, {0xc00007e980, 0x1, 0x8}, 0x0)
    /go/pkg/mod/github.com/coredns/caddy@v1.1.1/caddy.go:661 +0x5f6
github.com/coredns/caddy.ValidateAndExecuteDirectives({0x2142d18, 0xc000222e80}, 0xc000222e80, 0x0)
    /go/pkg/mod/github.com/coredns/caddy@v1.1.1/caddy.go:612 +0x3e5
github.com/coredns/caddy.startWithListenerFds({0x2142d18, 0xc000222e80}, 0xc00021c900, 0x0)
    /go/pkg/mod/github.com/coredns/caddy@v1.1.1/caddy.go:515 +0x274
github.com/coredns/caddy.Start({0x2142d18, 0xc000222e80})
    /go/pkg/mod/github.com/coredns/caddy@v1.1.1/caddy.go:472 +0xe5
github.com/coredns/coredns/coremain.Run()
    /go/src/coremain/run.go:63 +0x1cd
main.main()
    /go/src/coredns.go:12 +0x17
```

Reply:
yep, totally makes sense, you're right. will try to fix it today